### DISCUSSION TITLE
Ported to a Raspberry Pi Pico

### DISCUSSION THREAD

**Discussion by sayatone** [asb2m10/dexed#311] [Q&A]
I have ported this amazing code to a Raspberry Pi Pico microcontroller with some custom audio output circuitry.  
I can over clock the microcontroller but still can only get about 3 voice polyphony.   I would like to try optimize the code for the microcontroller by removing some of the other CPU intensive operations but fail to understand the inner workings of the code.    I noticed that a lot of trig lookup tables are being used for performance but don't understand the math/logic behind some of the other elements like the Freqlut, for example.   My audio circuit only uses a 12bit DAC so I don't need such high precision.  
If someone could point me to some additional documentation about the code so I can attempt to optimize it, it would be appreciated.  If I have missed some references that are already listed on this page or in the code comments, please accept my apologies in advance.

**Comment by Andy2No** [comment]
Have you looked at  dukesrg / logue-osc ?  https://github.com/dukesrg/logue-osc

I'm not sure how closely related to Dexed it is, but the FMxx oscillator is another STM32 based DX7 software emulation, targetted at the Korg Minilogue XD and Korg NTS-1.  He referred to Dexed quite often in the gearspace thread where he talked about developing FMxx, and he seems to know it intimately.

The thread is here:
https://gearspace.com/board/electronic-music-instruments-and-electronic-music-production/1306032-korg-logue-user-oscillator-programming-thread.html

That's still active, so you could ask him there, rather than posting questions on his github repository. He goes by the username dukesrg in that forum, and also on Github.  His real name is Oleg Burdaev, IIRC.  

If you jump to the last page, you should see some posts by him - currently the last page is page 78, and he has about a dozen posts on it:

https://gearspace.com/board/electronic-music-instruments-and-electronic-music-production/1306032-korg-logue-user-oscillator-programming-thread-78.html
